.gemini/styleguide.md

```diff
@@ -0,0 +1,715 @@
+- `self.run_layer_test()` - For individual layers
+- `self.run_preprocessor_test()` - For preprocessors
+- `self.run_task_test()` - For task models
+- `self.run_preset_test()` - For testing preset loading
+- `self.run_model_saving_test()` - For testing serialization
+
+### Example Test Structure
+```python
+import pytest
+from keras import ops
+
+from keras_hub.src.models.my_model.my_model_backbone import MyModelBackbone
+from keras_hub.src.tests.test_case import TestCase
+
+
+class MyModelBackboneTest(TestCase):
+    def setUp(self):
+        self.init_kwargs = {
+            "vocabulary_size": 10,
+            "num_layers": 2,
+            "hidden_dim": 16,
+            "intermediate_dim": 32,
+        }
+        self.input_data = {
+            "token_ids": ops.ones((2, 5), dtype="int32"),
+            "padding_mask": ops.ones((2, 5), dtype="int32"),
+        }
+
+    def test_backbone_basics(self):
+        self.run_backbone_test(
+            cls=MyModelBackbone,
+            init_kwargs=self.init_kwargs,
+            input_data=self.input_data,
+            expected_output_shape=(2, 5, 16),
+        )
+
+    @pytest.mark.large
+    def test_saved_model(self):
+        self.run_model_saving_test(
+            cls=MyModelBackbone,
+            init_kwargs=self.init_kwargs,
+            input_data=self.input_data,
+        )
+
+    @pytest.mark.large
+    def test_smallest_preset(self):
+        self.run_preset_test(
+            cls=MyModelBackbone,
+            preset="my_model_base",
+            input_data=self.input_data,
+            expected_output_shape=(2, 5, 16),
+            expected_partial_output=ops.array([1.0, 2.0, 3.0, 4.0, 5.0]),
+        )
+
+    @pytest.mark.extra_large
+    def test_all_presets(self):
+        for preset in MyModelBackbone.presets:
+            self.run_preset_test(
+                cls=MyModelBackbone,
+                preset=preset,
+                input_data=self.input_data,
+            )
+```
+
+## Validation Colab Notebooks
+
+### Requirement
+Each pull request must include links to Colab notebooks that demonstrate numerical equivalence with the original model's implementation.
+
+### Structure
+Provide separate Colabs for each major component:
+
+1. **Backbone Validation**: Load original weights into your KerasHub backbone and show that for the same input tensor, the output tensor is numerically identical (or within a very small tolerance).
+
+2. **Converter/Preprocessor Validation**: Show that your preprocessor pipeline produces the same token IDs, padding masks, or pixel values as the original library's preprocessing functions.
+
+3. **End-to-End Validation**: Use `MyModelTask.from_preset()` to load your pre-trained model and run a full task (e.g., classification). The final output (e.g., logits, probabilities) must match the original model.
+
+## Import Conventions
+
+### Keras Imports
+Prefer importing `keras` as a top-level object:
```

Comment:
`from keras import ops` is used a lot though

Reply:
oh yeah... removed this